scipy/signal/_spectral_py.py

```diff
@@ -646,21 +646,33 @@ def spectrogram(x, fs=1.0, window=('tukey', .25), nperseg=None, noverlap=None,
         `False` return a two-sided spectrum. Defaults to `True`, but for
         complex data, a two-sided spectrum is always returned.
     scaling : { 'density', 'spectrum' }, optional
-        Selects between computing the power spectral density ('density')
-        where `Sxx` has units of V**2/Hz and computing the power
-        spectrum ('spectrum') where `Sxx` has units of V**2, if `x`
-        is measured in V and `fs` is measured in Hz. Defaults to
-        'density'.
+        Selects between normalizing the window to unit power (``'density'``) or
+        unit area (``'spectrum'``). See the `mode` parameter for more
+        information. Defaults to ``'density'``.
```

Comment:
This only has an effect when `mode='psd'`, right? If so, it's worth adding that fact here so that it's immediately clear that using a different `mode` means that this param will be ignored.
```suggestion
        Selects between normalizing the window to unit power (``'density'``) or
        unit area (``'spectrum'``). Only used when ``mode='psd'``.
        See the `mode` parameter for more information.
        Defaults to ``'density'``.
```

Reply:
In [spectral_helper (line 1839)](https://github.com/scipy/scipy/blob/6405523c9967c11b108e396ad7faaf83ef526aff/scipy/signal/_spectral_py.py#L1839) the normalization is always changed regardless of the `mode` value. That makes the outcome a little bit hard to predict as shown [in this comment](https://github.com/scipy/scipy/issues/14903#issuecomment-1100249704).
IMHO only the combination ``scaling='density', mode='psd'`` as wells as ``scaling='spectrum', mode='magnitude'`` are the only combinations which useful in practice. For the other modes, it is probably better to  use the  `stft()` function directly.
Added a clarification. Thanks.